src/irmin-pack/pack.ml

```diff
@@ -205,6 +202,12 @@ struct
 
     let create = Lwt_mutex.create ()
 
+    let valid t =
```

Comment:
That might be safer not to have a side effect here, and just check for validity.

Reply:
All `valid` functions in `irmin-pack` also increase the counter, so maybe it's more consistent this way ?